src/Raven.Client/Documents/Session/Tokens/IntersectToken.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Text;
+
+namespace Raven.Client.Documents.Session.Tokens
+{
+    public class IntersectToken : QueryToken
```

Comment:
Better to use method syntax, instead:

`WHERE intersect( Name = 'John', Age > 20)`